packages/@react-aria/interactions/src/utils.ts

```diff
@@ -118,8 +118,10 @@ export function useSyntheticBlurEvent<Target = Element>(onBlur: (e: ReactFocusEv
       stateRef.current.observer = new MutationObserver(() => {
         if (stateRef.current.isFocused && target.disabled) {
           stateRef.current.observer.disconnect();
-          target.dispatchEvent(new FocusEvent('blur'));
-          target.dispatchEvent(new FocusEvent('focusout', {bubbles: true}));
+          // added `relatedTarget: document.activeElement` to keep focus on activeElment for Firefox DatePicker closing popover with minValue
```

Comment:
```suggestion
          // added relatedTarget because we have checks which specifically look for that property in order to determine if focus was lost to the body or intentionally moved with in the scope
```